### DISCUSSION TITLE
Release rollback which contains database migrations via git revert

### DISCUSSION THREAD

**Discussion by dro-sh** [werf/werf#5749] [General]
Let's suppose pipeline which automatically create new release on new git commit via CI/CD system. We have a release which change database schema via migration library (like [go-migrate](https://github.com/golang-migrate/migrate)). Migration apply successfully, but an application has some bug and we want to rollback release, which assumes apply down migration files to return database scheme before release. We revert latest commit via git revert. But reverting commit delete down migration files. Are there any recommendations for returning back database scheme on rollback release which has a database migrations?

**Comment by ilya-lesikov** [comment]
Hi, the fully automated solution might be something like this:
1. In the master CI pipeline you check whether the `migrations/` directory changed, compared to the previous deployed commit.
2. If `migrations/` changed, then try to detect to which database version you should roll back (maybe execute some commands on HEAD and previously deployed commit, or just detect which files are missing).
3. When you know to which version you should rollback, spawn new CI job in the same master pipeline which should execute something like `rake db:rollback <version>` on the previously deployed commit (execute in shell or as a Kubernetes job). The job must run before the application deployment job.

Alternatively, you can manually trigger the db rollbacks, e.g. add something like `[rollback db to <version>]` to your merge commit message and check for this in your master pipeline. If it's there, then spawn the job with `rake db:rollback <version parsed from commit message>` in your master pipeline.
